modular_skyrat/modules/modular_items/code/modular_glasses.dm

```diff
@@ -0,0 +1,237 @@
+  mode = modes[mode] // Change modes
+  to_chat(user, span_notice("[modes_msg[mode]][src].")) /// Print the message defined for this mode in the user's chat
+  switch(mode)
+    if(MODE_ON)
+      /// Resets all the vars to their initial values (THIS PRESUMES THE DEFAULT STATE IS ON)
+      worn_icon = initial(worn_icon)
+      icon_state = initial(icon_state)
+      flash_protect = initial(flash_protect)
+      tint = initial(tint)
+      vision_flags = initial(vision_flags)
+      hud_type = initial(hud_type)
+      hud_trait = initial(hud_trait)
+      clothing_traits = initial(clothing_traits)
+      ADD_TRAIT(user, hud_trait, GLASSES_TRAIT)
+    if(MODE_FREEZE_ANIMATION)
+      /// Create new icon and worn_icon, with only the first frame of every state and setting that as icon.
+      /// this practically freezes the animation :)
+      var/icon/I = new(icon, frame = 1)
+      icon = I
+      var/icon/W = new(worn_icon, frame = 1)
+      worn_icon = W
+  if(mode == MODE_OFF || mode == MODE_OFF_FLASH_PROTECTION) /// pass both off modes to this step
+    icon_state = off_state /// Sets icon_state to be the off variant set in the vars
+    flash_protect = (mode == MODE_OFF_FLASH_PROTECTION) ? FLASH_PROTECTION_FLASH : FLASH_PROTECTION_NONE /// when off is supposed to have flash protection
+    tint = (mode == MODE_OFF_FLASH_PROTECTION) ? 0 : 1 /// when off is suppost to tint the glasses
+    vision_flags = 0 /// Sets vision_flags to 0 to disable meson view mainly
+    lighting_alpha = user.default_lighting_alpha() // Resets lighting_alpha to user's default one
+    hud_type = null /// no hud when off, obviously!
+    hud_trait = null /// no hud when off, obviously!
+    clothing_traits = null /// also disables the options for Science functionality
+    REMOVE_TRAIT(user, hud_trait, GLASSES_TRAIT)
+  playsound(src, modeswitch_sound, 50, TRUE) // play sound set in vars!
+  // Blah blah, fix vision and update icons
+  if(ishuman(user))
+    var/mob/living/carbon/human/H = user
+    if(H.glasses == src)
+      H.update_sight()
+  update_action_buttons()
+  update_appearance()
+  ..()
+
+/obj/item/clothing/glasses/hud/ar/attack_self(mob/user)
+  toggle_mode(user, TRUE)
+
+/obj/item/clothing/glasses/hud/ar/aviator
+  name = "aviators"
+  desc = "A pair of designer sunglasses with electrochromatic darkening lenses!"
+  worn_icon = 'modular_skyrat/modules/modular_items/icons/modular_glasses_mob.dmi'
+  icon_state = "aviator"
+  icon = 'modular_skyrat/modules/modular_items/icons/modular_glasses.dmi'
+  darkness_view = 1
+  flash_protect = FLASH_PROTECTION_FLASH
+  modes = list(MODE_OFF = MODE_ON, MODE_ON = MODE_OFF)
+  tint = 0
+  off_state = "aviator_off"
+
+/obj/item/clothing/glasses/hud/ar/aviator/suicide_act(mob/living/carbon/user)
+  user.visible_message(span_suicide("[user] is posing rapidly with their [src] on... Clearly breathtaking. It looks like [user.p_theyre()] trying to commit suicide!"))
+  return OXYLOSS
```

Comment:
We have the suicide verb disabled, this will never be usable. Best to just not include it.